src/test/java/com/github/bibsysdev/nva/cristin/person/CristinPersonsHandlerTest.java

```diff
@@ -0,0 +1,227 @@
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.bibsysdev.nva.cristin.person.model.cristin.CristinPerson;
+import com.github.bibsysdev.nva.cristin.person.model.nva.NvaPerson;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import no.unit.nva.testutils.HandlerRequestBuilder;
+import nva.commons.apigateway.ApiGatewayHandler;
+import nva.commons.apigateway.GatewayResponse;
+import nva.commons.core.Environment;
+import nva.commons.core.JsonUtils;
+import nva.commons.core.ioutils.IoUtils;
+import org.apache.commons.codec.Charsets;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.mockito.Mockito;
+
+public class CristinPersonsHandlerTest {
+
+    private static final String LANGUAGE_KEY = "language";
+    private static final String EMPTY_STRING = "";
+    private static final String LANGUAGE_NB = "nb";
+    private static final String LANGUAGE_INVALID = "invalid";
+    private static final String NAME_MATTIAS = "Mattias";
+    private static final String INVALID_JSON = "This is not valid JSON!";
+    private static final String EMPTY_LIST_STRING = "[]";
+    private static final String ALLOW_ALL_ORIGIN = "*";
+    private static final String NVA_ONE_PERSON_JSON_FILE = "nva_one_person.json";
+    private static final String NVA_PERSONS_NON_ENRICHED_JSON_FILE = "nva_persons_non_enriched.json";
+    private static final String NVA_PERSONS_NO_HITS_JSON_FILE = "nva_persons_no_hits.json";
+    private static final String CRISTIN_ONE_PERSON_JSON_FILE = "cristin_one_person.json";
+    private final ObjectMapper objectMapper = JsonUtils.objectMapper;
+    private final Environment environment = new Environment();
+    private CristinApiClient cristinApiClientStub;
+    private Context context;
+    private ByteArrayOutputStream output;
+    private CristinPersonsHandler handler;
+
+    @Test
+    public void handlerReturnsOkWhenInputContainsName() throws Exception {
+        GatewayResponse<PersonsWrapper> response = sendDefaultQuery();
+        assertEquals(HttpURLConnection.HTTP_OK, response.getStatusCode());
+    }
+
+    @Test
+    public void handlerIgnoresErrorsWhenTryingToEnrichPersonInformation() throws Exception {
```

Comment:
When does the handler return a non-enriched person and when an enriched? What is the difference between an enriched person and a non-enriched?

Reply:
The cristin api returns a list of persons containing only their names and id's, basically a list of person summary objects. The specification for our api want's a list of full persons objects with all of the data you would expect to see when getting a single person. Therefore, our API has to do a full get on every person it finds in a query. The nva-cristin-project project called this process of gathering and adding more data/metadata than what cristin returns by default "enriching".

Reply:
It will always try to do this, but will return the basic info gathered if enriching fails.